### PR TITLE
some cleanup / small feature additions

### PR DESCRIPTION
**Pull request by rgarcia**
This PR does a handful of stuff that will be useful to have before converting something to regularly use this CLI:

- make sure we stay within limits on API calls to make sure we minimize the chance of these calls failing
- add ability to interpolate environment variables in command-line arguments, so we can pull in things like `_DEPLOY_ENV` in the activity name
- add ability to use ECS task ID as worker name, which will be useful for pulling logs for a workflow
- Parse input first as `[]string`, but fall back to a JSON object of any form

### CODE REVIEW

File: cmd/sfncli/sfncli.go [hunk 1]
```diff
@@ -65,28 +75,28 @@ func main() {
...
+	// a nil error and empty output. So wrap it in a polling loop of our own
 	ticker := time.NewTicker(5 * time.Second)
-OuterLoop:
+GetActivityTaskPollingLoop: // breaking out of for-select is not pretty :-/ https://golang.org/ref/spec#Break_statements
 	for {
 		select {
 		case <-mainCtx.Done():
 			log.Info("getactivitytask-stop")
-			break OuterLoop // :-/ https://golang.org/ref/spec#Break_statements
+			break GetActivityTaskPollingLoop
```
Comment (xavi-) [inline cmd/sfncli/sfncli.go]: Nit.  Possible alternatives:
- Use `return`
- Create an `isDone` flag and use it as conditional `for !isDone {...`

**Comment by xavi-** [comment]
Does this fix https://clever.atlassian.net/browse/INFRA-2526?

**Comment by rgarcia** [comment]
@xavi- mostly... it does not do the workflow ID stuff. Getting a workflow ID from an activity is not possible through the API (see AWS support ticket I opened about it), so if we want it, we would need to create a convention around adding it to the input. One strategy for doing that might be injecting a `_workflow_id` field into the input for a workflow in `workflow-manager`, parsing it in `sfncli`, logging it at the CLI level, and then injecting it to the command env for further logging. This would only work for workers whose input is a top-level JSON object, though. If a worker has input this is a top level JSON array then I'm not sure how we could create a convention around injecting additional input data.

**Comment by rgarcia** [comment]
The convention above gets messier actually... `sfncli` would also need to add `_workflow_id` into the activity output, otherwise subsequent workers won't have access to it :-/

File: cmd/sfncli/sfncli.go (hunk 1 above)
Comment (rgarcia) [reply cmd/sfncli/sfncli.go]: used `Err()` method on the context to get rid of label break here 4e5829c

File: cmd/sfncli/sfncli.go [hunk 2]
```diff
@@ -33,10 +34,22 @@ func main() {
...
+	*activityName = os.ExpandEnv(*activityName)
+	if *workerName == "" {
+		fmt.Println("workername is required")
+		os.Exit(1)
+	}
+	*workerName = os.ExpandEnv(*workerName)
+	if newWorkerName, err := expandECSTaskARN(*workerName); err != nil {
+		fmt.Printf("error expanding %s: %s", magicECSTaskARN, err)
+		os.Exit(1)
+	} else {
```
Comment (mohit) [inline cmd/sfncli/sfncli.go]: we don't really need this else but good either way :)

File: cmd/sfncli/sfncli.go [hunk 3]
```diff
@@ -116,7 +125,8 @@ OuterLoop:
 				log.InfoD("heartbeat-end", logger.M{"token": token})
 			}()
 
-			// Run the command
+			// Run the command. Treat unprocessed args (flag.Args()) as additional args to
```
Comment (mohit) [inline cmd/sfncli/sfncli.go]: 💯

File: README.md [hunk 4]
```diff
@@ -4,7 +4,26 @@ Utility to create AWS Step Function activities out of command line programs.
...
+Usage of sfncli:
+  -activityname string
+    	The activity name to register with AWS Step Functions. $VAR and ${VAR} env variables are expanded.
+  -cmd string
+    	The command to run to process activity tasks.
+  -region string
+    	The AWS region to send Step Function API calls. Defaults to AWS_REGION.
+  -version
+    	Print the version and exit.
+  -workername string
```
[the remark this answers is not in the document]
Comment (rgarcia) [reply README.md]: no, it's worker name as sent in the call to `GetActivityTask`: http://docs.aws.amazon.com/step-functions/latest/apireference/API_GetActivityTask.html

**Comment by rgarcia** [comment]
@mohit yeah, we should add a JIRA for handling signals. FWIW right now the default heartbeat timeout in states language is 60 seconds, so a job would fail 60 seconds after killing `sfncli`. The benefit of signal handling would be to make the failure happen immediately, and also add context that the failure was signaled

File: cmd/sfncli/sfncli.go (hunk 2 above)
Comment (rgarcia) [reply cmd/sfncli/sfncli.go]: it restricts the scope of `newWorkerName` to these 7 lines of code :stuck_out_tongue_winking_eye:

**Pull request merged**